### PR TITLE
chore: move universal-wallet dep to peerdeps

### LINKED ISSUE #127: Should @transmute/universal-wallet be a peer dependency?
Hullo — this might be my misunderstanding of the way you're supposed to extend things using `factory.ts` and the plugin pattern used in a bunch of these wallet repos, buuut...

## Use cases
I want to define my own composition root for my wallet [like this one](https://github.com/w3c-ccg/universal-wallet-interop-spec/blob/master/packages/universal-wallet-react/src/__fixtures__/testWalletFactory.ts) so that I can add more plugins.

~~I want to expose my own [`Wallet`](https://github.com/w3c-ccg/universal-wallet-interop-spec/blob/d1c044923a9fd51c108776c58d1fd31c39b6e1fa/packages/universal-wallet/src/walletFactory.ts#L13-L25)-based interface to other parts of my app, not [`SidetreeWallet`](https://github.com/transmute-industries/sidetree.js/blob/4d952c9cda501295da5cb0765c3ed8ba65e770e7/packages/wallet/src/types/SidetreePlugin.ts#L53), so I can isolate the rest of my app from changes occurring to this Sidetree's wallet interface.~~
I no longer think this one is valid because I wasn't thinking in TypeScript. I can just define my own interface which happens to match.

## Issue
`@transmute/universal-wallet` is added as a [`dependency`](https://github.com/transmute-industries/sidetree.js/blob/4d952c9cda501295da5cb0765c3ed8ba65e770e7/packages/wallet/package.json#L35).

## Fix
Add it as a `peerDependency`.

## Workaround
Install a matching version of `@transmute/universal-wallet` into my app.

### PR DESCRIPTION
**Pull request by NickDarvey**
Fixes #127

### CODE REVIEW

**Comment by gjgd** [comment]
Hi @NickDarvey , thanks for the PR!

I ran the tests on your branch and it looks like `npm run build` and `npm test` are failing in the `wallet package`.

I suspect adding `@transmute/universal-wallet` to `devDependencies` should fix the issue, because we use this lib in the tests

File: packages/wallet/package.json [hunk 1]
```diff
@@ -29,10 +29,12 @@
     "@types/base64url": "^2.0.0",
     "tsdx": "^0.13.3"
   },
+  "peerDependencies": {
+    "@transmute/universal-wallet": "^0.4.1"
```
Comment (gjgd) [inline packages/wallet/package.json:33]: add to `devDependencies` as well

**Review by gjgd — CHANGES_REQUESTED** [review]

File: packages/wallet/package.json (hunk 1 above)
Comment (NickDarvey) [reply packages/wallet/package.json:34]: * added to devDependencies
* ran `npm install` again
* ran `npm run build`
* ran `npm test`, 17/17 passing

**Review by gjgd — APPROVED** [review]
All good now!

This PR highlights a problem in our CI: since it relies on Github secrets it doesn't run on PRs from external contributors. I created https://github.com/transmute-industries/sidetree.js/issues/129 to track this issue

Thanks, approved!

**Pull request merged**